ocaml/libs/http-lib/http.ml

```diff
@@ -208,15 +208,6 @@ let parse_keyvalpairs xs =
       )
     kvpairs
 
-let parse_uri x =
-  match Astring.String.cuts ~sep:"?" x with
-  | [uri] ->
-      (uri, [])
-  | [uri; params] ->
-      (uri, parse_keyvalpairs params)
```

Comment:
Can we replace or remove `parse_keyvalpairs` too? It has just one other caller below.

Reply:
Done! 

Uri fits this use case too:
```ocaml
utop # let t = Uri.of_string "?k1=v1&k2=v2" in Uri.query t;;
- : (string * string list) list = [("k1", ["v1"]); ("k2", ["v2"])]
```

This threw away `urldecode`, `Malformed_url`, and `implode` as well.

Reply:
Oops, I take it back:

```ocaml
utop # let t = Uri.of_string "k1=v1&k2=v2" in Uri.query t;;
- : (string * string list) list = []
```

Reply:
I'm wondering how the only usage of `parse_keyvalpairs` works at all, since cookies are supposed to be split by `;`, not `&` as `parse_keyvalpairs` assumes. [MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cookie), confirmed by a random request done on Firefox. Do we control the other side and consider ampersands as delimiters?